### PR TITLE
Loading spinner for people component

### PR DESCRIPTION
**Pull request by dhutaryan**

### CODE REVIEW

File: src/app/people/people.component.ts [hunk 1]
```diff
@@ -14,10 +14,12 @@ import { Person } from '../models/person';
 export class PeopleComponent implements OnInit {
 
   people: Observable<Person[]>;
+  showSpinner: boolean = true;
 
   constructor(private HttpService: PeopleService) { }
 
   ngOnInit() {
     this.people = this.HttpService.getPeople().map(data => data.results);
+    this.people.subscribe(() => this.showSpinner = false);
```
Comment (dbl7) [inline src/app/people/people.component.ts]: what if request falls?

File: src/app/people/people.component.html [hunk 2]
```diff
@@ -20,6 +20,7 @@ <h1>Star Wars <small>Encyclopedia</small></h1><hr/>
               <tr app-person *ngFor="let person of people | async" [person]="person"></tr>
             </tbody>
           </table>
+          <app-spinner *ngIf="showSpinner"></app-spinner>
```
Comment (dbl7) [inline src/app/people/people.component.html]: Consider implementing spinner for all page, that will disable user interaction. And in order to use it inject service:
```javascript
spinnerService.enable();
spinnerService.disable();
```

File: src/app/people/people.component.ts (hunk 1 above)
Comment (dbl7) [reply src/app/people/people.component.ts]: You create a new subscription and it's not connected with your iteration in template

File: src/app/people/people.component.ts [hunk 3]
```diff
@@ -19,7 +19,9 @@ export class PeopleComponent implements OnInit {
   constructor(private HttpService: PeopleService) { }
 
   ngOnInit() {
-    this.people = this.HttpService.getPeople().map(data => data.results);
-    this.people.subscribe(() => this.showSpinner = false);
+    this.people = this.HttpService.getPeople().map(data => {
+      this.showSpinner = false;
```
Comment (dbl7) [inline src/app/people/people.component.ts]: It's not a good idea to handle loading state in `map` function, avoid adding side effect

File: src/app/people/people.component.ts [hunk 4]
```diff
@@ -19,7 +19,9 @@ export class PeopleComponent implements OnInit {
   constructor(private HttpService: PeopleService) { }
 
   ngOnInit() {
-    this.people = this.HttpService.getPeople().map(data => data.results);
-    this.people.subscribe(() => this.showSpinner = false);
+    this.people = this.HttpService.getPeople().map(data => {
```
Comment (dbl7) [inline src/app/people/people.component.ts]: If you declare `Observable` variable, please add `$` at the end of variable name

File: src/app/people/people.component.ts [hunk 5]
```diff
@@ -19,7 +19,9 @@ export class PeopleComponent implements OnInit {
   constructor(private HttpService: PeopleService) { }
```
Comment (dbl7) [inline src/app/people/people.component.ts]: Why did you named `PeopleService ` as `HttpService `?

File: src/app/people/people.component.ts (hunk 4 above)
Comment (dbl7) [reply src/app/people/people.component.ts]: Move `map` to your service, don't modify data in component, just invoke `getPeople` method

**Review by dbl7 — CHANGES_REQUESTED** [review]

File: src/app/people/people.component.ts (hunk 4 above)
Comment (dhutaryan) [reply src/app/people/people.component.ts]: Done

File: src/app/people/people.component.ts (hunk 5 above)
Comment (dhutaryan) [reply src/app/people/people.component.ts]: Fixed

File: src/app/people/people.component.ts (hunk 1 above)
Comment (dhutaryan) [reply src/app/people/people.component.ts]: Already fixed in the next commit

File: src/app/people/people.component.html [hunk 6]
```diff
@@ -17,10 +17,10 @@ <h1>Star Wars <small>Encyclopedia</small></h1><hr/>
               </tr>
             </thead>
             <tbody>
-              <tr app-person *ngFor="let person of people | async" [person]="person"></tr>
+              <tr app-person *ngFor="let person of people$ | async" [person]="person"></tr>
             </tbody>
           </table>
-          <app-spinner *ngIf="showSpinner"></app-spinner>
+          <!-- <app-spinner *ngIf="showSpinner"></app-spinner> -->
```
Comment (dbl7) [inline src/app/people/people.component.html]: Don't comment unnecessary code, if you don't need it just remove

File: src/app/people/people.service.ts [hunk 7]
```diff
@@ -1,16 +1,18 @@
...
 
 @Injectable()
 export class PeopleService {
 
   constructor(private http: HttpClient) { }
 
-  getPeople(): Observable<People> {
-    return <Observable<People>>this.http.get('https://swapi.co/api/people/');
+  getPeople(): Observable<Person[]> {
+    return <Observable<Person[]>>this.http.get('https://swapi.co/api/people/')
```
Comment (dbl7) [inline src/app/people/people.service.ts]: I believe you don't need to specify type `<Observable<Person[]>>` here, if it's needed just do it in the next line in map function

File: src/app/app.module.ts [hunk 8]
```diff
@@ -7,14 +7,16 @@ import { AppComponent } from './app.component';
...
 
 
 @NgModule({
   declarations: [
     AppComponent,
     PeopleComponent,
     PersonComponent,
-    SpinnerComponent
+    SpinnerComponent,
+    GenderPipe
```
Comment (dbl7) [inline src/app/app.module.ts]: Add a comma at the end

File: src/app/people/person/gender.pipe.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    if (gender === 'female') {
+      return 'Female';
+    }
+
+    if (gender === 'n/a') {
+      return 'Not available';
+    }
+  }
```
Comment (dbl7) [inline src/app/people/person/gender.pipe.ts]: Consider avoiding so many conditional statements

File: src/app/people/person/gender.pipe.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'gender'
```
Comment (dbl7) [inline src/app/people/person/gender.pipe.ts]: Comma 🙂

File: src/app/people/people.component.ts [hunk 11]
```diff
@@ -16,9 +17,9 @@ export class PeopleComponent implements OnInit {
   people$: Observable<Person[]>;
   showSpinner: boolean = true;
 
-  constructor(private HttpService: PeopleService) { }
+  constructor(private PeopleService: PeopleService) { }
```
Comment (dbl7) [inline src/app/people/people.component.ts]: Why does your variable start with uppercase letter?

File: src/app/people/people.component.ts [hunk 12]
```diff
@@ -2,6 +2,7 @@ import { Component, OnInit } from '@angular/core';
 import { PeopleService } from './people.service';
 import { Observable } from 'rxjs/Observable';
 import 'rxjs/add/operator/map';
+import 'rxjs/add/';
```
Comment (dbl7) [inline src/app/people/people.component.ts]: What do you need it for?

**Review by dbl7 — CHANGES_REQUESTED** [review]
Implement spinner indicator before merging

File: src/app/people/people.component.ts (hunk 12 above)
Comment (dhutaryan) [reply src/app/people/people.component.ts]: Fixed

File: src/app/people/people.component.ts (hunk 11 above)
Comment (dhutaryan) [reply src/app/people/people.component.ts]: Fixed

File: src/app/app.component.html [hunk 13]
```diff
@@ -1 +1,2 @@
-<app-people></app-people>
\ No newline at end of file
+<app-people></app-people>
+<app-spinner *ngIf="showSpinner"></app-spinner>
```
Comment (dbl7) [inline src/app/app.component.html]: Move this logic to spinner component

File: src/app/people/people.component.ts [hunk 14]
```diff
@@ -14,12 +14,23 @@ import { Person } from '../models/person';
 })
 export class PeopleComponent implements OnInit {
 
-  people$: Observable<Person[]>;
-  showSpinner: boolean = true;
+  people$: Person[];
```
Comment (dbl7) [inline src/app/people/people.component.ts]: Is it `Observable`?

File: src/app/shared/spinner/spinner.component.ts [hunk 15]
```diff
@@ -1,15 +1,21 @@
...
 export class SpinnerComponent implements OnInit {
+  showSpinner: boolean;
 
-  constructor() { }
+  constructor(private spinnerService: SpinnerService) { }
 
   ngOnInit() {
+    this.spinnerService.status.subscribe((value: boolean) => {
+      this.showSpinner = value;
+    });
```
Comment (dbl7) [inline src/app/shared/spinner/spinner.component.ts]: What do you need this logic for?

**Review by dbl7 — CHANGES_REQUESTED** [review]

File: src/app/people/person/gender.pipe.ts (hunk 10 above)
Comment (dhutaryan) [reply src/app/people/person/gender.pipe.ts]: Fixed

File: src/app/people/people.component.ts (hunk 14 above)
Comment (dhutaryan) [reply src/app/people/people.component.ts]: For future implementation. Fixed

File: src/app/app.component.html (hunk 13 above)
Comment (dhutaryan) [reply src/app/app.component.html]: Done

File: src/app/shared/spinner/spinner.component.ts (hunk 15 above)
Comment (dhutaryan) [reply src/app/shared/spinner/spinner.component.ts]: Now this logic is needed because spinner logic is moved from app component to spinner component.

**Pull request merged**